adc/consumer.py

```diff
@@ -154,6 +154,14 @@ def _stream_until_eof(self,
             try:
                 for m in messages:
                     err = m.error()
+                    # A new message may arrive from a previously removed topic/partition,
+                    # in which case it must be re-added
+                    if m.topic() not in active_partitions:
```

Comment:
How about using collections.defaultdict to avoid the explicit check here?

Reply:
Also, if we're adding topics dynamically, is the code around line 147 still neeeded?

Reply:
In the current structure, it is needed to be able to enter the main consumption loop. I suppose it would be possible to do some sort of `do. . . while` like iteration with a condition which is initially different, but I think there would also be an increased danger that with `active_partitions` initially empty we would start out reading a chunk of messages from just one partition (which we would auto-add), and if that chunk reached the end of that partition we would then leave the loop before noticing any messages from other partitions.

Reply:
I think it's good enough the way it is -- I was just curious if it's a 5-second fix or not. LGTM, go ahead and merge when ready.

Reply:
Switched to `defaultdict`.